### PR TITLE
Correctly suggest std or core path depending if this is a `no_std` crate

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
A few lints emit suggestions using `std` paths whether or not this is a `no_std` crate, which is an issue when running `rustfix` afterwards. So in case this is an item that is defined in both `std` and `core`, we need to check if the crate is `no_std` to emit the right path.

r? @llogiq 

changelog: Correctly suggest std or core path depending if this is a `no_std` crate

### CODE REVIEW

File: clippy_lints/src/methods/unnecessary_sort_by.rs [hunk 1]
```diff
@@ -204,7 +204,7 @@ pub(super) fn check<'tcx>(
             cx,
             UNNECESSARY_SORT_BY,
             expr.span,
-            "use Vec::sort_by_key here instead",
+            "consider using `sort_by_key`",
```
Comment (GuillaumeGomez) [inline clippy_lints/src/methods/unnecessary_sort_by.rs:207]: This method comes from slices so doesn't make much sense to only write `Vec::`.

**Comment by llogiq** [comment]
I wonder if a helper method that returns `std` outside `no_std` code and its argument (which could be "core" or "alloc") inside would markedly simplify the code here.

Otherwise the change looks good to me.

**Comment by GuillaumeGomez** [comment]
I thought about it as well, but I'm not sure what's the best course of action here for the arguments. Few ideas I had:

```
std_core_path("cmp::Ord") -> "core::cmp::Ord"
std_core_path(&["cmp", "Ord"]) -> "core::cmp::Ord"
std_core_path(&["std", "cmp", "Ord"]) -> "core::cmp::Ord"
```

A preference maybe?

**Comment by llogiq** [comment]
Since most usages will be used in formatting anyway, a function that returns either `"std"` or the given argument should cater for all uses.

**Comment by GuillaumeGomez** [comment]
Let's go for `get_std_core_prefix`.

**Comment by GuillaumeGomez** [comment]
Oh there is already `std_or_core`. Let's use this one then!

**Comment by llogiq** [comment]
Thank you!

@bors r+

**Pull request merged**